server/lib/scene/scene.get.js

```diff
@@ -58,14 +59,29 @@ async function get(options) {
   }
 
   if (optionsWithDefault.searchTags) {
-    const tags = await db.TagScene.findAll({
-      fields: 'scene_id',
-      where: {
-        [Op.or]: optionsWithDefault.searchTags.split(',').map((tag) => ({ name: { [Op.like]: `%${tag}%` } })),
-      },
+    const tags = optionsWithDefault.searchTags.split(',');
+
+    const sceneIdsAndNames = (
+      await db.TagScene.findAll({
+        fields: ['name', 'scene_id'],
+        where: {
+          [Op.or]: tags.map((tag) => ({ name: { [Op.like]: `%${tag}%` } })),
+        },
+      })
+    ).map((tag) => tag.get({ plain: true }));
+
+    const tagsWithSceneId = {};
+    sceneIdsAndNames.forEach((sceneIdAndName) => {
+      if (sceneIdAndName.name in tagsWithSceneId === false) {
+        tagsWithSceneId[sceneIdAndName.name] = [];
+      }
+      tagsWithSceneId[sceneIdAndName.name].push(sceneIdAndName.scene_id);
```

Comment:
No need to do all that, you need to do a JOIN with a simple WHERE :) 

In Sequelize: https://sequelize.org/docs/v6/advanced-association-concepts/eager-loading/#eager-loading-filtered-at-the-associated-model-level

Here you'll need to do n JOIN (one per tag)